### PR TITLE
Add checkM wrappers

### PR DESCRIPTION
**Pull request by bebatut**
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

I created a [spreadsheet to summarize inputs/outputs of the different commands](https://docs.google.com/spreadsheets/d/19OYS21o-qyVg5CgJnKnPSiP-4q1vix1NdcYB0xVPhj0/edit?usp=sharing). I am just not sure about the best way to add that to the help section

### CODE REVIEW

File: tools/checkm/plot.xml [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+#else if $plot.command == 'marker_plot'
+mkdir -p 'inputs/storage/' &&
+cp '$marker_gene_stats' 'inputs/storage/marker_gene_stats.tsv' &&
+cp '$bin_stats_ext' 'inputs/storage/bin_stats_ext.tsv' &&
+#for $b in $plot.genes
+    #set $identifier = re.sub('[^\s\w\-\\.]', '_', str($b.element_identifier))
+mkdir -p 'inputs/bins/${identifier}' &&
+cp '$b.file_name' 'inputs/bins/${identifier}/genes.faa' &&
+#end for
+ls 'inputs/bins' &&
```
Comment (bgruening) [inline tools/checkm/plot.xml:154]: can be removed?

File: tools/checkm/plot.xml [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+    --height $height
+
+#else if $plot.command == 'marker_plot'
+mkdir -p 'inputs/storage/' &&
+cp '$marker_gene_stats' 'inputs/storage/marker_gene_stats.tsv' &&
+cp '$bin_stats_ext' 'inputs/storage/bin_stats_ext.tsv' &&
+#for $b in $plot.genes
+    #set $identifier = re.sub('[^\s\w\-\\.]', '_', str($b.element_identifier))
+mkdir -p 'inputs/bins/${identifier}' &&
+cp '$b.file_name' 'inputs/bins/${identifier}/genes.faa' &&
```
Comment (bgruening) [inline tools/checkm/plot.xml:152]: ln -s does not work here?

File: tools/checkm/plot.xml [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+    --dpi $dpi
+    --font_size $font_size
+    --width $width
+    --height $height
+    --fig_padding $plot.fig_padding
+
+#else if $plot.command == 'gc_bias_plot'
+mkdir 'mapping' &&
+ln -s '$bam_file' 'mapping.bam' &&
+samtools index 'mapping.bam' 'mapping.bam.bai' &&
```
Comment (bgruening) [inline tools/checkm/plot.xml:170]: how big are those files? should we use multiple cores here?

File: tools/checkm/plot.xml [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+            <param name="height" value="3.5"/>
+            <output_collection name="marker_plot" count="1">
+                <element name="637000110" ftype="png">
+                    <assert_contents>
+                        <has_size value="137394" delta="10"/>
+                    </assert_contents>
+                </element>
+            </output_collection>
+        </test>
+        <!--<test expect_num_outputs="1">
```
Comment (bgruening) [inline tools/checkm/plot.xml:503]: What is with this test?

File: tools/checkm/qa.xml [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="biotools"/>
+    <expand macro="requirements"/>
+    <expand macro="version"/>
+    <command detect_errors="exit_code"><![CDATA[
+#import re
+mkdir -p 'output/storage/' &&
+cp '$checkm_hmm_info' 'output/storage/checkm_hmm_info.pkl.gz' &&
```
Comment (bgruening) [inline tools/checkm/qa.xml:14]: no linking?

File: tools/checkm/qa.xml [hunk 6]
```diff
@@ -0,0 +1,500 @@
...
+    --e_value $e_value
+    --length $length
+#if $coverage 
+    --coverage_file '$coverage'
+#end if
+    --threads \${GALAXY_SLOTS:-1}
+    ]]></command>
+    <inputs>
+        <expand macro="marker_file" />
+        <param name="checkm_hmm_info" type="data" format="zip" label="Marker gene HMM info for each bin" help="Output of the analyze command" />
```
Comment (bgruening) [inline tools/checkm/qa.xml:52]: Output of the "checkm analyze" tool.

File: tools/checkm/qa.xml [hunk 7]
```diff
@@ -0,0 +1,500 @@
...
+            <param argument="--out_format" type="select" label="Desired output">
+                <option value="1">Summary of bin completeness and contamination</option>
+                <option value="2">Extended summary of bin statistics (includes GC, genome size, ...)</option>
+                <option value="3">Summary of bin quality for increasingly basal lineage-specific marker sets</option>
+                <option value="4">List of marker genes and their counts</option>
+                <option value="5">List of bin id, marker gene id, gene id</option>
+                <option value="6">List of marker genes present multiple times in a bin</option>
+                <option value="7">List of marker genes present multiple times on the same scaffold</option>
+                <option value="8">List indicating position of each marker gene within a bin</option>
+                <option value="9">FASTA file of marker genes identified in each bin</option>
```
Comment (bgruening) [inline tools/checkm/qa.xml:65]: file__s__?

File: tools/checkm/qa.xml [hunk 8]
```diff
@@ -0,0 +1,500 @@
...
+4. ist of marker genes for each bin along with the number of times each marker was identified
+    Node Id: unique id of internal node in genome tree from which lineage-specific markers were inferred
+    Marker lineage: indicates lineage used for inferring marker set
+    Useful for identifying lineage-specific gene loss or duplication
+5. List of bin id, marker gene id, and called gene id for each identified marker gene
+6. List of marker genes present multiple times in a bin
+7. List of marker genes present multiple times on the same scaffold
+    Useful for identifying true gene duplication events, gene calling errors, or assembly errors. See note below.
+8. List indicating the position of each marker genes within a bin
+9. FASTA file of marker genes identified in each bin
```
Comment (bgruening) [inline tools/checkm/qa.xml:496]: file__s__

File: tools/checkm/taxonomy_wf.xml [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+    --aai_strain $qa.aai_strain
+    $qa.ignore_thresholds
+    --e_value $qa.e_value
+    --length $qa.length
+    --file '$results'
+    --tab_table
+    --extension 'fasta'
+    --threads \${GALAXY_SLOTS:-1}
+&&
+ls output/
```
Comment (bgruening) [inline tools/checkm/taxonomy_wf.xml:34]: remove?

File: tools/checkm/format_taxon_list.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    # create a table per rank
+    for rank in taxonomicRanks:
+        rank_fp = tool_data_dp / Path("%s.loc.sample" % rank)
+        with rank_fp.open('w') as rank_f:
+            rank_f.write('# File generated by format_taxon_list.py script\n')
+            rank_f.write('# taxon\tdescription\n')
+            for taxon in sorted(taxonMarkerSets[rank]):
+                markerSet = taxonMarkerSets[rank][taxon]
+                numMarkers, numMarkerSets = markerSet.size()
+                rank_f.write('%s\t%s (%s genomes, %s marker genes, %s marker sets)\n' % (taxon, taxon, markerSet.numGenomes, numMarkers, numMarkerSets))
```
[the remark this answers is not in the document]
Comment (bebatut) [reply tools/checkm/format_taxon_list.py:22]: Thanks for the suggestion

File: tools/checkm/plot.xml (hunk 1 above)
Comment (bebatut) [reply tools/checkm/plot.xml:154]: Yes. Removed...

File: tools/checkm/plot.xml (hunk 2 above)
Comment (bebatut) [reply tools/checkm/plot.xml:152]: I tried with `ln -s` but it did not work... I did not really investigate

File: tools/checkm/plot.xml (hunk 3 above)
Comment (bebatut) [reply tools/checkm/plot.xml:170]: Good question... I only used a test data. But this option is currently disabled as I could not make it work properly (and not really the highest priority in this tool)

File: tools/checkm/plot.xml (hunk 4 above)
Comment (bebatut) [reply tools/checkm/plot.xml:503]: It is a test for the `gc_bias_plot` option, which I disabled for now (see previous comment)

File: tools/checkm/qa.xml (hunk 5 above)
Comment (bebatut) [reply tools/checkm/qa.xml:14]: Same as a previous comment. `ln -s` did not work...

File: tools/checkm/taxonomy_wf.xml (hunk 9 above)
Comment (bebatut) [reply tools/checkm/taxonomy_wf.xml:34]: Thanks. I forgot to remove...

File: tools/checkm/qa.xml (hunk 8 above)
Comment (bebatut) [reply tools/checkm/qa.xml:496]: It is not really a FASTA file. It is a tabular file including sequences

File: tools/checkm/qa.xml (hunk 7 above)
Comment (bebatut) [reply tools/checkm/qa.xml:65]: It is not really a FASTA file. It is a tabular file including sequences

**Review by bgruening — APPROVED** [review]

**Pull request merged**